flexmeasures_client.py

```diff
@@ -180,13 +194,14 @@ def handle_response_errors(self, message, res, description, fnc, **fnc_kwargs):
 
 # TODO AJO 2022-02-26: dit zou in fm_ha_module moeten zitten...
 def search_for_kwh_target(description: Optional[str]) -> Optional[int]:
-    """Search description for the first occurrence of some (integer) number of kWh.
+    """Search description for the first occurrence of some (integer) number of %.
 
     Forgives errors in incorrect capitalization of the unit and missing/double spaces.
     """
     if description is None:
         return None
-    match = re.search("(?P<quantity>\d+) *kwh", description.lower())
+    pattern = re.compile(r"(?P<quantity>\d+) *%")
+    match = pattern.search(description.lower())
```

Comment:
You can't really capitalize `%`, so the call to `lower` has now become confusing. I would find it preferable to move the choice of unit into the function signature, so you can reuse this function. You could then even consider making the choice of unit a user setting (and load that config setting not within this function, but just before calling this function, or even earlier, when initializing the app).

Reply:
Good suggestion. Maybe make it even more robust/simple: no matter what input is make this (% or kWh) return a quantity in kWh. And i think that kWh is more specific and should be the leading one if both would be present.
It would then require that this function receives the max SoC in kWh.

Reply:
It would probably be cleaner (more modular) to keep that logic outside of this function, perhaps by making a separate util function that calls this function once with `unit="kWh"` and another time with `unit="%"`.

Hint: also use `unit.lower()` when inserting it in the regex.

Reply:
Done!